### PR TITLE
Shields now have weakpoints

### PR DESCRIPTION
**Pull request by Super3222**
Revisions:
- Projectiles now carry the direction the firer was facing.
- IsReflect() now carrys two mobs. For the person defending and the person firing.
- Shields now have weak points and go through a check depending on the direction of the attacker.

**Red arrows indicate where projectiles will be able to land on the target**
**Blue arrows indicate where projectiles will be blocked/reflected by the target**

This covers **ONLY** riot shields.

To-do:
- ~~Make disarming depend on your position.~~
- ~~Make thrown items check for positions.~~

Special thanks to KorPhaeron who guided me through how lists like 'blockcheck' operate, a long, long time ago.

:cl: Super3222
tweak: Riot Shields and their telescopic versions now have blindspots where projectiles/attacks/thrown items will land on them from directly behind
/:cl:

### CODE REVIEW

**Comment by FluffySurvivor** [comment]
this is a nerf

**Comment by ShadowDeath6** [comment]
this makes ops even less likely to buy eshields, :-1:

**Comment by Super3222** [comment]
Oh please, Shadow. If nukeops won't bother to attempt to block projectiles then they shouldn't bother buy a shield at all.

**Comment by ShadowDeath6** [comment]
They might as well always buy the shielded hardsuit since it's more taser proof than the shield specifically designed to block tasers

**Comment by Super3222** [comment]
Oh please, Shadow. If nukeops aren't going to attempt to block projectiles then they shouldn't bother buy a shield at all.

**Comment by Super3222** [comment]
Energy Shields and Shidled Hardsuits have their own pros and cons. This isn't to outwiegh the usefulness of either of those items specifically, this is a change to how combat works regarding shields.

**Comment by ShadowDeath6** [comment]
> this won't outweigh the usefulness of eshield

yes it will, because if any autismo mcgrayshirt with a taser can run up behind you while you're busy killing captain and shoot you in the back, completely nullifying 16 TC worth of items, people will stop buying it. I could care less about riot shields since they're annoying as all hell, but this creates another stupidly easy counter to a shield with one stupidly easy counter already. It's literally DESIGNED to block energy projectiles, and it can't do it as well as a shielded hardsuit that blocks bullets as well as energy.

**Comment by Super3222** [comment]
I'm sorry that you have to at least face the way the projectiles coming from?

**Comment by ShadowDeath6** [comment]
balance > realism, don't gimp eshields more than they already are

**Comment by ParagonPrime** [comment]
decrease the price of the eshield and this is good

**Comment by FluffySurvivor** [comment]
Super, you have to compensation buff it.

**Comment by ShadowDeath6** [comment]
or you could just not make it effect eshields ;^)

**Comment by Time-Green** [comment]
You can place shields on your backpack slot.
Could we actually make it block stuff that way?

**Comment by Super3222** [comment]
> You can place shields on your backpack slot.
> Could we actually make it block stuff that way?

I was actually thinking of that while coding this :P

**Comment by Stealthkibbler** [comment]
@FluffySurvivor or you could fucking get good and learn how to moonwalk block

**Comment by ShadowDeath6** [comment]
> spamclicking while trying to loot someone with the giant ass tgui popup that lags your game
> realism is fun

**Comment by Stealthkibbler** [comment]
@ShadowDeath6 comparing a 30 cost elite shielded hardsuit to a 16 cost double autismo blade, good one.

**Comment by Super3222** [comment]
Word of advice from this. Cover your weak points/blind spots, and you should be completely fine. If your reflexes are razor sharp then you'll be able to deflect anything at any time. 

Also... WALLHUGGING will become a trend. Now... ~~instead of complaining about how your speshful snowflaek shield now requires at least some level of skill~~ give me some time to polish a few things.

**Comment by ShadowDeath6** [comment]
Just remove the shield if it has so many counters that 16 TC is better spent elsewhere.

**Comment by Bekons** [comment]
This is shit. Remove the shield then.

Unless you will add something like mech strafing for wall huging, you make the best parts about both the items gone. You will have to buff there cost like mad then cause it's just plain not worth the cost.

Also, you let people DISARM someone when they have a double esword? Make there arm dismember for attempting to go hand to hand when the second person has a LIGHTSABER.

**Comment by Bekons** [comment]
And another thing.

You nerf antags way too much while buffing crew. Why?

Antags are supposed to be op shitstraters that they were made to be. Crew isn't supposed to 1v1 a nuke op. You nerf eshield to absolute uselessness but keep riot shield buffed? the riot shield can block projectiles and works 360. Why not change eshield with riot shield at this point on the uplinks?

**Comment by TheSadFox** [comment]
> This covers riot shields, telescopic shields, energy shields, and dual sabers.
> 
> You nerf eshield to absolute uselessness but keep riot shield buffed? the riot shield can block projectiles and works 360.

????

**Comment by ghost** [comment]
make this not cover dualswords and eshields or i'm shutting it down

**Comment by TheSadFox** [comment]
> make this not cover dualswords and eshields or i'm shutting it down

Bodypillows PR déjà vu.

**Comment by Super3222** [comment]
> You nerf antags way too much while buffing crew. Why?

Wait a minute? WHAT? I started out working on riot shields with this one, since it's been a long requested feature after ported KorPhaeron's shield overhaul / made sheilds breakable. This also applies on riot shields... and telescopic shields...

> You nerf eshield to absolute uselessness but keep riot shield buffed?

Bekons seriously, what are you talking about now? I specifically said that this covers riot shields, I'm pretty sure that's the first thing that's said. Maybe it's the picture I used that's misleading.

> the riot shield can block projectiles and works 360.

No it can't. And it doesn't work 360. **NOTHING** will work 360. My thought process wasn't 'xddd hey you know whats terrible? ESHIELDS! les make em only protect the front of you'.

No. I was working on riot shields, got riot shields done. I started working on telescopic shields, got them done. I was about to submit this PR. I realized there are still energy shields, so I thought about it for a moments notice. Why shouldn't I bring esheild's into this? Isn't it only fair that if we're creating blind spots where projectiles won't be blocked or reflected? So yeah, eshields got done. I then went over to dual sabers, despite them not being a shield they carried the same traits.

> Also, you let people DISARM someone when they have a double esword? Make there arm dismember for attempting to go hand to hand when the second person has a LIGHTSABER

That sounds like a fair trade honestly, and I've even been told a wielded dual esword dismembers on the first hit.

> make this not cover dualswords and eshields or i'm shutting it down

That sort of aggressiveness towards PR's is the reason why you can't close them anymore. Or if you meant you're planning on writing me a five paragraph essay on why I shouldn't use this on eshields and dual sabers, go ahead.

**Comment by ParagonPrime** [comment]
Stop bitching like little fucks.

Just because you're too fucking complacent to consider _strafing_ doesn't mean the rest of us have an issue with this.

Merge the PR, see what happens, if the nukeops win rate drops hugely because of it then revert it. It's really that simple.

Also, dual eswords should not have full coverage- even if eshields get full protection, no way in hell eswords.

As for making them disarmproof, I've supported that since day 1. Absolutely.

**Comment by TheSadFox** [comment]
Like the changes, but honestly don't think double blades should lose the 360 protection, would be too much of a nerf imho. Eshields are usually used with ranged weapons in another hand, which easly nullifies any danger from baton wielding crewmembers, along with allowing you to safely position yourself and face the direction from which they won't be able to pierce the shield with tasers from the unprotected side, and if someone with ballistics arrives you should have enough distance between you and them to safely retreat. 
Swords on the other hand, you cannot access any ranged weapons unless you disable the sword, which also disables reflection, put you in the danger zone of a hard counter in form of ggbaton, and with all the dancing and spamclicking that happens during melee fights, i believe it will be incredibly difficult to block the shots from an unprotected side. 

As for price reduction for an eshield if this nerf goes through, sure, why not?

Also Shadow, eshield is a handicap for bad players, and if you use it as an ops outside of war, you should be ashamed of yourself.

**Comment by ghost** [comment]
yes super i meant essay

**Comment by Stealthkibbler** [comment]
Just lower the cost or some shit and people won't complain

**Comment by Super3222** [comment]
You raise a good point, Seiko.

**Comment by Super3222** [comment]
Turns out I was wrong about something;


Only the middle arrow is the actual weakpoint (behind the HoS). I'll look into this more.

EDIT: It basically depends on where you're firing from. If you are directly behind the guy with the shield WHILE FIRING, then it will hit them. However, if you aren't then the shield will deal with it.

**Comment by ShadowDeath6** [comment]
the problem with riot shields is that in normal rounds they negated the most common source of damage in normal rounds

eshields have two super easy counters and if you make the one thing they're supposed to block able to be countered if you run around the person and spam shots it makes them ass

**Comment by TheSadFox** [comment]
> eshields have two super easy counters and if you make the one thing they're supposed to block able to be countered if you run around the person and spam shots it makes them ass

Batons and ballistics. Batons you have to be really bad at the game to get dunked with while you are using ops ranged weapons, unless they are on meth or something. Ballistics, the only one readily available is dets revolver. Usually if the ops decide to use eshields they will go for the armory and just space all the riot shotguns there, or just do their job fast enough to not allow security to even aquire them from. Thats not really a lot of super easy counters.

**Comment by Super3222** [comment]
In the last commit:
- Dual Energy Swords no longer respect positions. Go check out Seiko's earlier response. It'd be truly unfair to fit it in the big "shield overhaul" as well.
- In respect for the "incredibly overpowering nerf" energy shields cost was reduced from 16 -> 14
- Thrown Items now respect positions
- Disarming now respects positions. Meaning you can one-up the HoS with his back turned.
- Check my earlier response for updates towards the weak/blind point situation

**Comment by ShadowDeath6** [comment]
> batons

all melee, actually, especially thrown spears that you can't un-embed through your hardsuit

> ballistics are rare and hard to find

I can guarantee if you're in war, "the only time ops should get eshields", people are going to buy a fuckton of shotguns to kill ops, especially now that cargo gets so many points from crates.

**Comment by Stealthkibbler** [comment]
I'm pretty sure WAR doesnt exist anymore, I got OP and there was no declare war button afaik.

**Comment by Cruix** [comment]
War is still an option, but only the leader gets to choose.

**Comment by Kmc2000** [comment]
Can we not, eshields are an essential for ops, and knowing the skill level on yogs, this is the wrong server to nerf eshields on

**Comment by TheSadFox** [comment]
> eshields are an essential for ops

:laughing:

**Comment by ShadowDeath6** [comment]
there's nothing essential for ops, but they should be able to choose what fits their playstyle, and making one item complete shit with 3 hard counters will make everything else seem much better to spend TC on

**Comment by TheSadFox** [comment]
> complete shit

:laughing:

**Comment by ShadowDeath6** [comment]
> :laughing:

:laughing:

**Comment by Super3222** [comment]
Look ... look.. guys calm down. Alright, everybody calm? I guess not because I can't exactly see your faces.

This is a simple test, just like Paragon said, this'll be a simple test merge. If it doesn't fit and Yogstation players are as unrobust as you guys say they are, then **I will personally revert this**.

If we get no complaints, nuclear operative wins and losses stay the same then I will even higher up the price on this weapon.

Alright? Everybody happy? This is not permanent if it doesn't fit and if it doesn't fit it will be reverted.

**Comment by ShadowDeath6** [comment]
doubt anything will change if the ops just decide not to buy the shields, might be a better idea to track who wants to spend the TC on it

**Comment by Alek2ander** [comment]
Why would you use the attacker's dir for anything? You can just check the direction of the projectile when it hits the person being shot. Remember, projectiles can be reflected and redirected.

**Comment by Super3222** [comment]
@Alek2ander 
For melee and thrown attacks, because checking the object/weapons dir will make it get buggy (object dir's do not correspond with where the person is, even when thrown).

I'm pretty sure I went with projectiles first before checking where the attacker's dir was facing and it didn't work out. I'll check it again nevertheless.

**Comment by Alek2ander** [comment]
If not the projectile's direction then the side the projectile/attack is coming from. Pretty sure the initial Bump() for the projectile is called while it's still on the tile neighboring the person being hit, so its loc can be used to determine the direction of the attack.

**Comment by Super3222** [comment]
After speaking with Oisin, it's been made clear to me that I don't need to set eshield players on fire right after they've already been badly nerfed.

**Comment by ShadowDeath6** [comment]
> :^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^)

merg dis

**Comment by FluffySurvivor** [comment]
keep teleshield unnerfed, it is already rare and it deserves to be strong

riot shields yes, they are spammable right off the bat so no mercy for them

**Comment by ShadowDeath6** [comment]
teleshields are spammable if R&D gets any materials
fuck em

**Comment by X-TheDark** [comment]
Teleshields need Combat 6 though now. And Cargo usually doesn't order shotgun crates outside of Blob rounds.

File: code/game/objects/items/weapons/shields.dm [hunk 1]
```diff
@@ -2,6 +2,30 @@
...
+#define IMPROPERBLOCK 0
+#define PROPERBLOCK 1
+
+var/global/list/blockcheck = list("[NORTH]" = list("[SOUTH]" = PROPERBLOCK, "[EAST]" = PROPERBLOCK, "[WEST]" = PROPERBLOCK, "[NORTH]" = IMPROPERBLOCK),
+"[EAST]" = list("[SOUTH]" = PROPERBLOCK, "[WEST]" = PROPERBLOCK, "[EAST]" = IMPROPERBLOCK, "[NORTH]" = PROPERBLOCK),
+"[SOUTH]" = list("[NORTH]" = PROPERBLOCK, "[WEST]" = PROPERBLOCK, "[EAST]" = PROPERBLOCK, "[SOUTH]" = IMPROPERBLOCK ),
+"[WEST]" = list("[NORTH]" = PROPERBLOCK, "[EAST]" = PROPERBLOCK, "[SOUTH]" = PROPERBLOCK, "[WEST]" = IMPROPERBLOCK) )
+
+
+/obj/item/weapon
```
Comment (oisin100) [inline code/game/objects/items/weapons/shields.dm]: Uhm, Mayaswell get rid of this and the comment below...

**Pull request merged**